src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1352,6 +1352,53 @@ impl Step for CrateRunMakeSupport {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct CrateBuildHelper {
+    host: TargetSelection,
+}
+
+impl Step for CrateBuildHelper {
+    type Output = ();
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/tools/build_helper")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(CrateBuildHelper { host: run.target });
+    }
+
+    /// Runs `cargo test` for build_helper.
+    fn run(self, builder: &Builder<'_>) {
+        let host = self.host;
+        let compiler = builder.compiler(builder.top_stage, host);
+
+        builder.ensure(compile::Std::new(compiler, host));
```

Comment:
Any reason for this?

Reply:
You mean this specific line or the whole step? I copy-pasted the whole step from the `RunMakeSupport` step.

Reply:
This specific line. `RunMakeSupport` is using specific compiler in its type so it makes sense to ensure std for it. But we don't need to build the top stage compiler and std for testing build helper crate, using stage 0 compiler should be enough (and faster as we don't need to build entire tree for testing it). Same for `CrateRunMakeSupport` step too, it should use stage 0 compiler without needing to build compiler and std.

Reply:
Ok, changed it to stage 0.